src/Commands/Plugin/PluginBaseCommand.php

```diff
@@ -0,0 +1,335 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Plugin;
+
+use Pantheon\Terminus\Commands\TerminusCommand;
+use Pantheon\Terminus\Exceptions\TerminusNotFoundException;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * Class PluginBaseCommand
+ * Base class for Terminus commands that deal with sending Plugin commands
+ * @package Pantheon\Terminus\Commands\Plugin
+ */
+abstract class PluginBaseCommand extends TerminusCommand
+{
+    /**
+     * PluginBaseCommand constructor.
+     */
+    public function __construct()
+    {
+    }
+
+    /**
+     * Get plugin projects.
+     *
+     * @param string $plugins_dir Plugins directory
+     * @return array Plugin projects
+     */
+    protected function getPluginProjects($plugins_dir)
+    {
+        $projects = array();
+        $finder = new Finder();
+        $finder->files()->in($plugins_dir);
+        foreach ($finder as $file) {
+            $path = $file->getRelativePath();
+            // Get the parent path only.
+            if (!strpos($path, '/')) {
+                // Make sure the path is unique.
+                if (!in_array($path, $projects)) {
+                    $projects[] = $path;
+                }
+            }
+        }
+        return $projects;
+    }
+
+    /**
+     * Get the plugin directory.
+     *
+     * @param string $plugin Plugin name
+     * @return string Plugin directory
+     */
+    protected function getPluginDir($plugin = '')
```

Comment:
@ronan: I appreciate the feedback.  The rub here is this function returns the entire path of the plugin, not just the `~/.terminus/plugins` directory.  I suppose we could still use this function as a wrapper around `$this->getConfig()->get('plugins_dir');` instead of including all the extra logic that determines the base directory.